src/cmd_line/commands/file.ts

```diff
@@ -32,109 +32,94 @@ export class FileCommand extends node.CommandBase {
   get arguments(): IFileCommandArguments {
     return this._arguments;
   }
-
-  getActiveViewColumn(): vscode.ViewColumn {
-    const active = vscode.window.activeTextEditor;
-
-    if (!active) {
-      return vscode.ViewColumn.One;
+  async execute(): Promise<void> {
+    if (this._arguments.bang) {
+      await vscode.commands.executeCommand('workbench.action.files.revert');
+      return;
     }
 
-    return active.viewColumn!;
-  }
-
-  getViewColumnToRight(): vscode.ViewColumn {
-    const active = vscode.window.activeTextEditor;
+    // Need to do this before the split since it looses the activeTextEditor
+    let editorFilePath = vscode.window.activeTextEditor!.document.uri.fsPath;
 
-    if (!active) {
-      return vscode.ViewColumn.One;
+    // Do the split if requested
+    let split = false;
+    if (this._arguments.position === FilePosition.NewWindowVerticalSplit) {
+      await vscode.commands.executeCommand('workbench.action.splitEditorRight');
+      split = true;
     }
-
-    switch (active.viewColumn) {
-      case vscode.ViewColumn.One:
-        return vscode.ViewColumn.Two;
-      case vscode.ViewColumn.Two:
-        return vscode.ViewColumn.Three;
+    if (this._arguments.position === FilePosition.NewWindowHorizontalSplit) {
+      await vscode.commands.executeCommand('workbench.action.splitEditorDown');
+      split = true;
     }
 
-    return active.viewColumn!;
-  }
-
-  async execute(): Promise<void> {
-    if (this._arguments.bang) {
-      await vscode.commands.executeCommand('workbench.action.files.revert');
-      return;
-    }
+    let hidePreviousEditor = async function() {
+      if (split === true) {
+        await vscode.commands.executeCommand('workbench.action.previousEditor');
+        await vscode.commands.executeCommand('workbench.action.closeActiveEditor');
+      }
+    };
 
+    // No name was specified
     if (this._arguments.name === undefined) {
-      // Open an empty file
-      if (this._arguments.position === FilePosition.CurrentWindow) {
+      // Should we create an empty file?
```

Comment:
Not sure why you left a question as a comment?

Reply:
It only reflected the underlying if condition. I have removed it.